init.go

```diff
@@ -1,17 +1,23 @@
 package config
 
 import (
+	"crypto/rand"
 	"encoding/base64"
 	"fmt"
 	"io"
 	"time"
 
+	"github.com/ipfs/interface-go-ipfs-core/options"
 	ci "github.com/libp2p/go-libp2p-core/crypto"
 	peer "github.com/libp2p/go-libp2p-core/peer"
 )
 
 func Init(out io.Writer, nBitsForKeypair int) (*Config, error) {
-	identity, err := identityConfig(out, nBitsForKeypair)
+	return InitWithOptions(out, []options.KeyGenerateOption{options.Key.Size(nBitsForKeypair)})
+}
+
+func InitWithOptions(out io.Writer, opts []options.KeyGenerateOption) (*Config, error) {
```

Comment:
We've wanted to be able to initialize configs with pre-existing identities before. What if we:

1. Changed this to `DefaultConfig(identity ci.PrivateKey)`
2. Added a `CreateIdentity(out io.Writer, opts... options.KeyGenerateOption) (ci.PrivateKey, error)` function.

This gives us a nice _pure_ function to construct configs from identities, and gives us a nice function to initialize new identities.